iambic/core/utils.py

```diff
@@ -144,14 +147,17 @@ async def gather_templates(repo_dir: str, template_type: str = None) -> list[str
         if template_type
         else NOQ_TEMPLATE_REGEX
     )
+    # since multiple glob pattern can potential intersect, we use a set data structure
+    # to suppress any duplicate for defensive measure
     # Support both yaml and yml extensions for templates
-    file_paths = glob.glob(f"{repo_dir}/**/*.yaml", recursive=True)
-    file_paths += glob.glob(f"{repo_dir}*.yaml", recursive=True)
-    file_paths += glob.glob(f"{repo_dir}/**/*.yml", recursive=True)
-    file_paths += glob.glob(f"{repo_dir}*.yml", recursive=True)
+    file_path_set = set()
+    # >>> glob.glob('**/*.txt', recursive=True)
+    # ['2.txt', 'sub/3.txt']
+    file_path_set.update(glob.glob(f"{repo_dir}/**/*.yaml", recursive=True))
```

Comment:
Again, would recommend using pathlib.

In pathlib you can do something like: list(p.glob('**/*.yaml')) or p.match('ya?ml'), which would match against the glob style pattern. https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.match